system/docker/run.sh

```diff
@@ -25,28 +26,81 @@ test_target="${1:-$DEF_TEST_TARGET}"
+elif [[ -f $test_target ]]; then
+    echo "$test_target is a file"
+    test_name=$(echo $test_target | sed -nr 's;.*tests/(.*).py.*;\1;p')
+    test_output_file="test-result-indy-test-automation-${test_name}.txt"
+else
+    echo "$test_target is not valid"
+    exit 1
+fi
+cd -
+
+
 repo_path=$(git rev-parse --show-toplevel)
 user_id=$(id -u)
 group_id=$(id -g)
-docker_socket_path="/var/run/docker.sock"
+
+# Set the following variables based on the OS:
+# - docker_socket_path
+# - docker_socket_mount_path
+# - $docker_socket_user_group
+. set_docker_socket_path.sh
+
 workdir_path="/tmp/indy-test-automation"
 
-image_repository="hyperledger/indy-test-automation"
-client_image_name="${image_repository}:client"
+image_repository=${IMAGE_REPOSITORY:="hyperledger/indy-test-automation"}
+client_image=${CLIENT_IMAGE:=":client-${UBUNTU_VERSION}"}
+client_image_name="${image_repository}${client_image}"
+node_image=${NODE_IMAGE:=":node-${UBUNTU_VERSION}"}
+node_image_name="${image_repository}${node_image}"
+ubuntu_version="${UBUNTU_VERSION}" 
 client_container_name="indy-test-automation-client"
 
-command_setup="
-    set -ex
-    pipenv --three
-    # We need this because pipenv installs the latest version of pip by default.
-    # The latest version of pip requires the version in pypi exactly match the version in package's setup.py file.
-    # But they don't match for python3-indy... so we need to have the old version of pip pinned.
-    pipenv run pip install pip==10.0.1
-    pipenv run pip install -r system/requirements.txt
+node_env_variables=" \
+    IMAGE_REPOSITORY \
+    UBUNTU_VERSION \
 "
+echo "Environment env variables..."
+for i in $node_env_variables
+do
+    echo "$i=${!i}"
+done
+
+if [[ "${ubuntu_version}" ==  "ubuntu-1604" ]]; then
+    command_setup="
+        set -ex
+        pip3 install --user pipenv
+        pipenv --three
+        # We need this because pipenv installs the latest version of pip by default.
+        # The latest version of pip requires the version in pypi exactly match the version in package's setup.py file.
+        # But they don't match for python3-indy... so we need to have the old version of pip pinned.
+        pipenv run pip install pip==10.0.1
+        pipenv run pip install -r system/requirements-ubuntu-1604.txt
+    "
+elif [[ "${ubuntu_version}" == "ubuntu-2004" ]]; then
+        command_setup="
+        set -ex
+        pip3 install --user pipenv
+        pipenv --three
+        # We need this because pipenv installs the latest version of pip by default.
+        # The latest version of pip requires the version in pypi exactly match the version in package's setup.py file.
+        # But they don't match for python3-indy... so we need to have the old version of pip pinned.
+        pipenv run pip install pip==10.0.1
+        pipenv run pip install -r system/requirements-ubuntu-2004.txt
+    "
+else
+    echo "Please, specify the version of ubuntu (ubuntu-1604|ubuntu-2004)"
+    exit 1
+fi
```

Comment:
This could be cleaned up.  The only difference here is the requirements file used for install.